### PR TITLE
Cooling Sensor Integration / 4024.35 Build Upgrade.

### PR DESCRIPTION
**Pull request by nrwslac**
## Description

- Renamed EL3054 terminals from default
- Upgraded lcls-twincat-motion from 2.0.1 -> 4.0.5
- Upgraded lcls-twincat-optics from 0.4.1 -> -0.6.1
- Added LCLS General 2.10.0
- Upgraded project build from 4022.* -> 4024.35 to match PLC run-time.
- Upgraded ADS version 0.2.1 -> 0.6.1
- Adds variable cooling PVs.
- Updates Cooling sensors to optics library implementation.
- Removed un-used EL6695. This project had original plans to control both the L and K beamlines and thus had 2 EL6695s. One was removed and moved to the lcls-plc-txi-hxr-optics project.
- This is henceforth the K line TXI optics PLC.
- Move main from GVL to PRG.

## Motivation and Context

- Cooling integration
- PLC run-time / build mismatch resolution

## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive comments
- [ ] Test suite passes locally
- [ ] Libraries are set to fixed versions and not ``Always Newest``
- [ ] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

File: lcls-plc-txi-optics/_Config/IO/Device 1 (EtherCAT)/Term 1 (EK1200).xti [hunk 1]
```diff
@@ -16,11 +16,8 @@
 		<Box File="Term 4 (EL6695).xti" Id="4">
 			<EtherCAT PortABoxInfo="#x01000003"/>
 		</Box>
-		<Box File="Term 5 (EL6695).xti" Id="5">
-			<EtherCAT PortABoxInfo="#x01000004"/>
-		</Box>
```
Comment (ZLLentz) [inline lcls-plc-txi-optics/_Config/IO/Device 1 (EtherCAT)/Term 1 (EK1200).xti:21]: Undocumented change here: term 5 removed. You should mention this in your PR desc with some justification.

File: lcls-plc-txi-optics/lcls-plc-txi-optics.tsproj [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+<?xml version="1.0"?>
+<TcSmProject xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.beckhoff.com/schemas/2012/07/TcSmProject" TcSmVersion="1.0" TcVersion="3.1.4024.35" TcVersionFixed="true">
 	<Project ProjectGUID="{650988AB-4830-4B99-8734-B5D72AA79AEE}" TargetNetId="172.21.88.54.1.1" ShowHideConfigurations="#x306">
 		<System>
+			<Licenses CacheOrCheckLicensesOnStartup="true">
+				<Target>
+					<ManualSelect>{3EBB9639-5FF3-42B6-8847-35C70DC013C8}</ManualSelect>
+					<LicenseDevice DongleHardwareId="2" DongleDevice="#x03020003" DongleLevel="50" DongleSystemId="{73EBFB0F-DFFE-1ECC-1159-7590D325BD5B}"/>
+				</Target>
+			</Licenses>
```
Comment (ZLLentz) [inline lcls-plc-txi-optics/lcls-plc-txi-optics.tsproj:10]: Undocumented change: setting up license dongle

File: lcls-plc-txi-optics/txi_optics/GVLs/Main.TcGVL
Comment (ZLLentz) [inline lcls-plc-txi-optics/txi_optics/GVLs/Main.TcGVL:1]: What's exactly going on here? It looks like these are copies from MAIN.TcPOU but unused?

**Review by ZLLentz — COMMENTED** [review]
Some undocumented changes that should be noted in the PR and one oddity
Mostly looks good

File: lcls-plc-txi-optics/_Config/IO/Device 1 (EtherCAT)/Term 1 (EK1200).xti (hunk 1 above)
Comment (nrwslac) [reply lcls-plc-txi-optics/_Config/IO/Device 1 (EtherCAT)/Term 1 (EK1200).xti:21]: Updated PR description to include the intentional removal of this terminal.

File: lcls-plc-txi-optics/lcls-plc-txi-optics.tsproj (hunk 2 above)
Comment (nrwslac) [reply lcls-plc-txi-optics/lcls-plc-txi-optics.tsproj:10]: I actually don't know what this is about. I don't recall adding a license dongle. I suppose I may have found the EL6070 unassigned and added in the license config.

File: lcls-plc-txi-optics/txi_optics/GVLs/Main.TcGVL
Comment (nrwslac) [reply lcls-plc-txi-optics/txi_optics/GVLs/Main.TcGVL:1]: > What's exactly going on here? It looks like these are copies from MAIN.TcPOU but unused?

Good question. Investigating.

Comment (nrwslac) [reply lcls-plc-txi-optics/txi_optics/GVLs/Main.TcGVL:1]: Main GVL was migrated to PRG MAIN. The Main GVL file wasnt fully deleted. Now removed.

Comment (ZLLentz) [reply lcls-plc-txi-optics/txi_optics/GVLs/Main.TcGVL:1]: Makes sense! (full disclosure: I'd like to refactor all of the projects to switch Main.TcGVL -> Main.TcPOU given the time)

File: lcls-plc-txi-optics/lcls-plc-txi-optics.tsproj (hunk 2 above)
Comment (ZLLentz) [reply lcls-plc-txi-optics/lcls-plc-txi-optics.tsproj:10]: sounds good to me

**Review by ZLLentz — APPROVED** [review]
Passes my review after those small changes!

**Pull request merged**